components/browser/storage-sync-places/src/main/java/mozilla/components/browser/storage/places/Conn.kt

```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.storage.places
+
+import org.mozilla.places.PlacesConnection
+
+const val DB_NAME = "places.sqlite"
+
+/**
+ * A singleton which holds a shared reference to [PlacesConnection].
+ */
+internal object Conn {
+    @Volatile
+    private var placesConnection: PlacesConnection? = null
```

Comment:
It seems we need to call `close` on `PlacesConnection`. Need to think about how we do this otherwise we'd leak every time the app process is killed/restarted.  Maybe expose a `close` method on `Conn`?

Reply:
@csadilek yup, good call. I've updated `HistoryStorage` to let us do that.